File: Starter Projects/Python/ConsequencesStarterClient/tests/SocketClientTests.py
import unittest

from SocketClient import SocketClient


class SocketClientTests(unittest.TestCase):
    def test_connect_to_server(self):
        # arrange
        socket_client = SocketClient("ws://51.141.52.52:1234")

        # act
        socket_client.send({"Hello": "Hello"})
        response = socket_client.receive()

        # assert
        self.assertEqual("Welcome to Consequences, to get started send your name and room code.", response.message)

File: Starter Projects/Java/ConsequencesStarterClient/src/main/java/tests/SocketClientTests.java
package tests;
import main;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertSame;

class SocketClientTests {

    @Test
    void demoTestMethod() {
        // Arrange
        SocketClient socketClient = new SocketClient("ws://51.141.52.52:1234");
        socketClient.Connect();

        // Act
        OutboundMessage message = new OutboundMessage();
        message.Hello = "Hello";
        socketClient.Send(message);
        var response = socketClient.Receive();

        // Assert
        assertSame("Welcome to Consequences, to get started send your name and room code.", response.Message);
    }
}
